src/main.cpp

```diff
@@ -204,6 +203,28 @@ String fetchDescriptor()
   return String("");
 }
 
+void sequence_motions(float newTarget) {
+  if (target == newTarget) {  // TODO: make it nicer
+    return;
+  }
+  timeline.clear();
+  timeline.add(target);
+  if (target < SERVO_MIDPOINT) {
+    if (newTarget < SERVO_MIDPOINT && target > newTarget) {
```

Comment:
Can you add a comment here about why there are two motion algorithms if the needle is either side of the midpoint?

Reply:
I'm not sure you need all these if statements. Can't you just replace the whole thing with:

```cpp
if (newTarget > SERVO_MIDPOINT && target < newTarget) {
  timeline[target].then<Ease::BounceOut>(newTarget, abs(newTarget - target) * ANIMATION_SPEED);
} else {
  timeline[target].then<Ease::BackInOut>(newTarget, abs(newTarget - target) * ANIMATION_SPEED * 2);
}
```

`abs(newTarget - target)` and `abs(target - newTarget)` are equivalent aren't they?

Reply:
I don't think so. The current algorithm only uses bounce when going from high to moderate or extreme to catastrophic, but your algorithm uses bounce when going from (either extreme or catastrophic) to (either moderate or high). (I think)
<img width="1238" alt="Screenshot 2024-11-28 at 10 45 14" src="https://github.com/user-attachments/assets/2fe6e3e9-0199-4f99-9f6f-5bbbafa6e025">
I think it might be because the servo at 0˚ is as far right as it will go, and 180˚ is as far left as it will go. Not entirely sure, though.